src/store/index.js

```diff
@@ -34,33 +40,36 @@ export default new Vuex.Store({
             sort: "rating"
           }
         });
-
         commit("setRestaurants", response.data.restaurants);
       } catch (error) {
         console.log(error.message);
       }
     },
 
-    //get one restaurant by ID
-    async fetchRestaurantById({ commit }, ID) {
-      try {
-        const response = await axios.get("/restaurant", {
-          params: {
-            res_id: ID
-          }
-        });
-
-        commit("setCurrentRestaurant", response.data);
-      } catch (error) {
-        console.log(error.message);
+    // get restaurant from cache or fetch it if not available
+    async fetchRestaurantFromCache({ commit, state }, ID) {
+      if (state.cachedRestaurantsByID.has(ID)){
+        commit("setRestaurant", state.cachedRestaurantsByID.get(ID));
+      } else {
+        try {
+          const response = await axios.get("/restaurant", {
+            params: {
+              res_id: ID
+            }
+          });
+          commit("setRestaurant", response.data);
+          commit("addToCache", response.data)
+        } catch (error) {
+          console.log(error.message);
+        }
```

Comment:
Now, if I go to the same "about" page twice, it's cached. However, when I go from the list of best restaurants to a single details page, it will still be reloaded. So it still takes N+1 queries, where N is the number of best restaurants listed on the main page.